File: src/main/java/february_2025/ContainsDuplicate.java
package february_2025;

import java.util.HashSet;
import java.util.Set;

public class ContainsDuplicate {

    public static void main(String[] args) {
        int[] nums = {1, 2, 3, 1};

        ContainsDuplicate containsDuplicate = new ContainsDuplicate();
        boolean result = containsDuplicate.containsDuplicate(nums);
        System.out.println(result);
    }

    // O(n) time | O(n) space
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> seen = new HashSet<>();

        for (int num : nums) {
            if (seen.contains(num)) {
                return true;
            }
            seen.add(num);
        }
        return false;
    }

}

File: python/contains_duplicate.py
class Solution:
    # O(n) time | O(n) space
    def containsDuplicate(self, nums):
        seen = set()

        for num in nums:
            if num in seen:
                return True
            seen.add(num)
        
        return False


nums = [1,1,1,3,3,4,3,2,4,2]
solution = Solution()

print(solution.containsDuplicate(nums))